### PR TITLE
Metrics: initialise views before meters to re-enable histograms

### PR DESCRIPTION
**Pull request by daravep**
Before this change, the whole histogram building logic was broken since we upgraded telemetry. Now, we initialize the view before the meters. As part of this change, I also move the documentation into the meter definition such that the metrics can include a description in otel, and finally I've improved the filtering such that we now also filter by qualification, which means that a client will not be overwhelmed with 400 unnecessary metrics.

Part of https://github.com/DACH-NY/canton/issues/17917

Note that this only does the minimum amount of changes to the Canton code, only to get it work with the new tooling, without removing the old & deprecated methods. Once we've moved Canton over to only use attribute based metrics, we'll have to come back to this and rip out the deprecated methods. (I still regret saying "i'll fix this quickly ... ")

### CODE REVIEW

File: sdk/canton/community/ledger/ledger-common/src/main/scala/com/digitalasset/canton/metrics/Metrics.scala [hunk 1]
```diff
@@ -8,18 +8,21 @@ import com.daml.metrics.api.MetricHandle.LabeledMetricsFactory
...
 import com.daml.metrics.api.opentelemetry.OpenTelemetryMetricsFactory
 import com.daml.metrics.grpc.DamlGrpcServerMetrics
+import com.typesafe.scalalogging.LazyLogging
 import io.opentelemetry.api.metrics.Meter
 import io.opentelemetry.sdk.metrics.SdkMeterProvider
 
-object Metrics {
+object Metrics extends LazyLogging {
+
+  lazy val KNOWN_METRICS = Set("booh")
```
Comment (nicu-da) [inline sdk/canton/community/ledger/ledger-common/src/main/scala/com/digitalasset/canton/metrics/Metrics.scala:17]: Do we wanna keep this?

File: sdk/observability/telemetry/src/main/scala/com/daml/telemetry/OpenTelemetryOwner.scala [hunk 2]
```diff
@@ -15,76 +16,76 @@ import scala.jdk.CollectionConverters.SeqHasAsJava
...
-        histogramSelectorByName(
-          s"${DamlGrpcServerMetrics.GrpcServerMetricsPrefix}*${OpenTelemetryTimer.TimerUnitAndSuffix}"
-        ),
-        explicitHistogramBucketsView(
-          Seq(
-            0.01d, 0.025d, 0.050d, 0.075d, 0.1d, 0.15d, 0.2d, 0.25d, 0.35d, 0.5d, 0.75d, 1d, 1.25d,
-            1.5d, 1.75d, 2d, 2.25d, 2.5d, 2.75d, 3d, 3.5d, 4d, 4.5d, 5d, 10d,
-          )
+      HistogramDefinition(
+        s"${DamlGrpcServerMetrics.GrpcServerMetricsPrefix}*${OpenTelemetryTimer.TimerUnitAndSuffix}",
```
Comment (nicu-da) [inline sdk/observability/telemetry/src/main/scala/com/daml/telemetry/OpenTelemetryOwner.scala:44]: Just double checking here, is there a way we can override this one? 
If not can we change the logic to register in such a way that it first goes through the config and if it doesn't find anything there then it goes through these provided candidates?

Reasoning:
We recently had a scenario where we wanted more detailed gRPC buckets as our latency was much higher than the max 10s and we wanted to see improvements there.

**Review by nicu-da — APPROVED** [review]
Nice, thank you!

File: sdk/canton/community/ledger/ledger-common/src/main/scala/com/digitalasset/canton/metrics/Metrics.scala (hunk 1 above)
Comment (daravep) [reply sdk/canton/community/ledger/ledger-common/src/main/scala/com/digitalasset/canton/metrics/Metrics.scala:17]: Yes. This is the same as for the canton metrics above. I'll finish that in the Canton PR.

**Pull request merged**